nemo/collections/asr/models/online_diarizer.py

```diff
@@ -0,0 +1,573 @@
+        """
+        Initialize an online segmentor module
+
+        Attributes:
+            online_segmentor (OnlineSegmentor):
+                online segmentation module that generates short speech segments from the VAD input
+        """
+        self.online_segmentor = OnlineSegmentor(sample_rate)
+
+    def _init_memory_buffer(self):
+        """
+        Variables are kept in memory for future updates
+
+        Attributes:
+            memory_margin (int):
+                The number of embeddings saved in the memory buffer.
+                This memory margin is dependent on the base scale length: margin = (buffer_length)/(base scale shift)
+                memory margin is automatically calculated to have minimal memory usage
+            memory_segment_ranges (dict):
+                The segment range information kept in the memory buffer
+            memory_segment_indexes (dict):
+                The segment indexes kept in the memory buffer
+            memory_cluster_labels (Tensor):
+                The cluster labels inferred in the previous diarization steps
+        """
+        self.memory_margin = 0
+        self.memory_segment_ranges = {key: [] for key in self.multiscale_args_dict['scale_dict'].keys()}
+        self.memory_segment_indexes = {key: [] for key in self.multiscale_args_dict['scale_dict'].keys()}
+        self.memory_cluster_labels = torch.tensor([])
+
+    def _init_temporal_major_voting_module(self):
+        """
+        Variables needed for taking majority votes for speaker labels
+
+        Attributes:
+            use_temporal_label_major_vote (bool):
+                Boolean for whether to use temporal majority voting
+            temporal_label_major_vote_buffer_size (int):
+                buffer size for majority voting
+            base_scale_label_dict (dict):
+                Dictionary containing multiple speaker labels for major voting
+                Speaker labels from multiple steps are saved for each segment index.
+        """
+        self.temporal_label_major_vote_buffer_size = 11
+        self.base_scale_label_dict = {}
+
+    def _init_segment_variables(self):
+        """
+        Initialize segment variables for each scale.
+        Note that we have `uniq_id` variable in case where multiple sessions are handled.
+        """
+        self.emb_vectors = {self.uniq_id: {}}
+        self.time_stamps = {self.uniq_id: {}}
+        self.segment_range_ts = {self.uniq_id: {}}
+        self.segment_raw_audio = {self.uniq_id: {}}
+        self.segment_indexes = {self.uniq_id: {}}
+
+        for scale_idx, (window, shift) in self.multiscale_args_dict['scale_dict'].items():
+            self.multiscale_embeddings_and_timestamps[scale_idx] = [None, None]
+            self.emb_vectors[scale_idx] = None
+            self.time_stamps[scale_idx] = []
+            self.segment_range_ts[scale_idx] = []
+            self.segment_raw_audio[scale_idx] = []
```

Comment:
why is `segment_raw_audio` initialized with `uniq_id` but fetched by `scale_idx` key?

Reply:
Fixed this.